### PR TITLE
add sounds

### PR DESCRIPTION
**Pull request by Nikkie-game-dev**

### CODE REVIEW

File: FlappyGame/src/Scenes/GamePlay.cpp [hunk 1]
```diff
@@ -2,11 +2,12 @@
 
 #include "GameLoop.h"
 
-#include "Entities/obstacle.h"
-#include "Entities/Player.h"
 #include "Entities/Parallax.h"
+#include "Entities/Player.h"
+#include "Entities/obstacle.h"
```
Comment (Stevendotcom) [inline FlappyGame/src/Scenes/GamePlay.cpp:7]: obstacle should be Obstacle

File: FlappyGame/src/Scenes/GamePlay.cpp [hunk 2]
```diff
@@ -28,12 +29,15 @@ namespace game::scenes::gameplay
...
 	float timerStart = 3.0f;
 	bool pause;
 
-	void InitEntities();
+	static bool wasOnTop = false;
+	static bool isOnTop = false;
+
+	static void InitEntities();
+
+	static void RestartEntities();
```
Comment (Nikkie-game-dev) [inline FlappyGame/src/Scenes/GamePlay.cpp]: static is not necessary

File: FlappyGame/src/Scenes/GamePlay.cpp (hunk 1 above)
Comment (Nikkie-game-dev) [inline FlappyGame/src/Scenes/GamePlay.cpp]: obstacle should be Obstacle

File: FlappyGame/src/Scenes/GamePlay.cpp [hunk 3]
```diff
@@ -64,21 +68,31 @@ namespace game::scenes::gameplay
 		if (timer > timerStart || !isMultiplayer)
 		{
 			if (IsKeyPressed(KEY_ESCAPE))
+			{
+				AddToBuffer(utils::soundManager::Sounds::PauseUp);
```
Comment (Nikkie-game-dev) [inline FlappyGame/src/Scenes/GamePlay.cpp]: use 'using namespace'

File: FlappyGame/src/Scenes/MainMenu.cpp [hunk 4]
```diff
@@ -3,20 +3,26 @@
...
 namespace game::scenes::mainmenu
 {
 	const std::string GameVersion = "Version 1.0";
-	constexpr int maxButtons = 4;
+	const int maxButtons = 4;
 
 	button::Button buttons[maxButtons];
 
+	static bool wasOnTop[maxButtons] = {false};
+	static bool isOnTop[maxButtons] = {false};
```
Comment (Nikkie-game-dev) [inline FlappyGame/src/Scenes/MainMenu.cpp]: idem

File: FlappyGame/src/Utilities/SoundManager.cpp [hunk 5]
```diff
@@ -0,0 +1,125 @@
+﻿#include "SoundManager.h"
+
+#include <array>
+#include <raylib.h>
+
+namespace game::utils::soundManager
+{
+	static const int bufferSpace = 10;
```
Comment (Nikkie-game-dev) [inline FlappyGame/src/Utilities/SoundManager.cpp]: idem

File: FlappyGame/src/Utilities/SoundManager.cpp [hunk 6]
```diff
@@ -0,0 +1,125 @@
...
+		}
+	}
+
+
+
+	void ChangeMusic(Musics nextMusic)
+	{
+		StopMusicStream(music);
+
+		switch (nextMusic)
```
Comment (Nikkie-game-dev) [inline FlappyGame/src/Utilities/SoundManager.cpp:68]: there should be a default here

File: FlappyGame/src/Utilities/SoundManager.cpp [hunk 7]
```diff
@@ -0,0 +1,125 @@
...
+	}
+
+
+
+	void AddToBuffer(Sounds sound)
+	{
+		newSoundPos++;
+		newSoundPos %= bufferSpace;
+
+		switch (sound)
```
Comment (Nikkie-game-dev) [inline FlappyGame/src/Utilities/SoundManager.cpp:92]: also here

**Review by Nikkie-game-dev — COMMENTED** [review]
a few small issues are to be fix

**Pull request merged**